packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing.types.ts

```diff
@@ -0,0 +1,343 @@
+  MANUAL = 'manualPayment',
+  EXPIRED = 'expired',
+  MANUAL_FORCED = 'forcedManual',
+  BILLING_SUSPENDED = 'billingSuspended',
+  UNKNOWN = 'unknown',
+}
+
+export type ServiceInfos = {
+  serviceId: number;
+  // date format yyyy-mm-dd
+  expiration: string;
+  canDeleteAtExpiration: boolean;
+  renew: {
+    period: Period;
+    automatic: boolean;
+    forced: boolean;
+    manualPayment: boolean;
+    deleteAtExpiration: boolean;
+  };
+  // date format yyyy-mm-dd
+  engagedUpTo: string | null;
+  domain: string;
+  contactTech: string;
+  status: 'expired' | 'inCreation' | 'ok' | 'pendingDebt' | 'unPaid';
+  // date format yyyy-mm-dd
+  creation: string;
+  contactBilling: string;
+  possibleRenewPeriod: Period[];
+  renewalType:
+    | 'automaticForcedProduct'
+    | 'automaticV2012'
+    | 'automaticV2014'
+    | 'automaticV2016'
+    | 'manual'
+    | 'oneShot'
+    | 'option';
+  contactAdmin: string;
+};
+
+export type Language =
+  | 'cs_CZ'
+  | 'de_DE'
+  | 'en_AU'
+  | 'en_CA'
+  | 'en_GB'
+  | 'en_IE'
+  | 'en_US'
+  | 'es_ES'
+  | 'fi_FI'
+  | 'fr_CA'
+  | 'fr_FR'
+  | 'fr_MA'
+  | 'fr_SN'
+  | 'fr_TN'
+  | 'it_IT'
+  | 'lt_LT'
+  | 'nl_NL'
+  | 'pl_PL'
+  | 'pt_PT';
+
+export type ContactInfos = {
+  address: {
+    country: string;
+    city: string;
+    zip: string;
+    line1: string;
+    line2?: string;
+    line3?: string;
+    otherDetails?: string;
+    province?: string;
+  };
+  birthCity?: string;
+  birthCountry?: string;
+  // Date
+  birthDay?: string;
+  birthZip?: string;
+  firstName: string;
+  // Contact id
+  id: number;
+  language: Language;
+  lastName: string;
+  legalForm:
+    | 'administration'
+    | 'association'
+    | 'corporation'
+    | 'individual'
+    | 'other'
+    | 'personalcorporation';
+  phone: string;
+  cellPhone?: string;
+  fax?: string;
+  gender?: 'male' | 'female';
+  companyNationalIdentificationNumber?: string;
+  nationalIdentificationNumber?: string;
+  nationality?: string;
+  organisationName?: string;
+  organisationType?: string;
+  spareEmail?: string;
+  vat?: string;
+};
+export type Strategy =
+  | 'CANCEL_SERVICE'
+  | 'REACTIVATE_ENGAGEMENT'
+  | 'STOP_ENGAGEMENT_FALLBACK_DEFAULT_PRICE'
+  | 'STOP_ENGAGEMENT_KEEP_PRICE';
+
+export type CurrencyCode =
+  | 'AUD'
+  | 'CAD'
+  | 'CZK'
+  | 'EUR'
+  | 'GBP'
+  | 'INR'
+  | 'LTL'
+  | 'MAD'
+  | 'N/A'
+  | 'PLN'
+  | 'SGD'
+  | 'TND'
+  | 'USD'
+  | 'XOF'
+  | 'points';
+
+export type LifecycleAction =
+  | 'earlyRenewal'
+  | 'terminate'
+  | 'terminateAtEngagementDate'
+  | 'terminateAtExpirationDate';
+
+export type ServiceDetails = {
+  billing: {
+    // Datetime format
+    nextBillingDate: string;
+    // Datetime format
+    expirationDate?: string;
+    plan: {
+      code?: string;
+      invoiceName?: string;
+    };
+    pricing: {
+      capacities: (
+        | 'consumption'
+        | 'detach'
+        | 'downgrade'
+        | 'dynamic'
+        | 'installation'
+        | 'renew'
+        | 'upgrade'
```

Comment:
**question:** What is the need to maintain this info?
Because the team doesn't have any control over the API response and it can change frequently which makes maintaining a difficult task.
For example, these are the possible values in Unified catalog but when we move to Commercial catalog, only `installation` and `renew` are possible.

Reply:
The benefits : autocompletion and typechecking with typescript
It is not crucial to maintain a precise implementation of the typing of the API but it gives good hints when coding so it is better to have this than just specifying `string` as a type.